757 add exaone path 2.0

cjs6211 opened the pull request:
Fixes #757  .

### Description
This PR adds the latest version of the EXAONE Path model, EXAONE Path 2.0.
The model is based on Hugging Face and is hosted at: https://huggingface.co/LGAI-EXAONE/EXAONE-Path-2.0.
### Status
**Ready**

### Please ensure all the checkboxes:
<!--- Put an `x` in all the boxes that apply, and remove the not applicable items -->
- [ ] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [ ] In-line docstrings updated.
- [ ] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [ ] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [ ] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [ ] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [X] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [X] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

cjs6211 committed:
Add files via upload

Add EXAONE Path 2.0

Signed-off-by: Jongseong Jang <37769148+cjs6211@users.noreply.github.com>

cjs6211 committed:
Update README.md

Signed-off-by: Jongseong Jang <37769148+cjs6211@users.noreply.github.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ericspod commented:
Hi @cjs6211 thank you for the contribution. We do require that models in our zoo use the [bundle format](https://docs.monai.io/en/stable/bundle_intro.html). You have a lot of your own code here so I wouldn't suggest reformatting everything to use MONAI components, but there does need to be at least an inference config file which can call into code you have defined. This [bundle here](https://github.com/ericspod/model-zoo/tree/dev/models/brain_image_synthesis_latent_diffusion_model) is an example of an inference script with few components, and just calls into a defined method to implement an inference process. You can follow that pattern here but please do look at other bundles and see what you can do to make your code into a bundle. 

We do also require that this model be in the `models` directory and not `hf_models` which contains information on those models hosted on Huggingface. We already reference a [EXAONEPath model hosted on hf](https://huggingface.co/LGAI-EXAONE/EXAONEPath) in `hf_models`,  if this is a later version of the same model perhaps it should be hosted in hf instead of here and we add a reference in `hf_models` in a similar pattern. 

@aylward could you please double check that this model's license is acceptable to us, I think it's fine from our previous discussions on licenses.

Thanks!

cjs6211 commented:
> Hi @cjs6211 thank you for the contribution. We do require that models in our zoo use the [bundle format](https://docs.monai.io/en/stable/bundle_intro.html). You have a lot of your own code here so I wouldn't suggest reformatting everything to use MONAI components, but there does need to be at least an inference config file which can call into code you have defined. This [bundle here](https://github.com/ericspod/model-zoo/tree/dev/models/brain_image_synthesis_latent_diffusion_model) is an example of an inference script with few components, and just calls into a defined method to implement an inference process. You can follow that pattern here but please do look at other bundles and see what you can do to make your code into a bundle.
> 
> We do also require that this model be in the `models` directory and not `hf_models` which contains information on those models hosted on Huggingface. We already reference a [EXAONEPath model hosted on hf](https://huggingface.co/LGAI-EXAONE/EXAONEPath) in `hf_models`, if this is a later version of the same model perhaps it should be hosted in hf instead of here and we add a reference in `hf_models` in a similar pattern.
> 
> @aylward could you please double check that this model's license is acceptable to us, I think it's fine from our previous discussions on licenses.
> 
> Thanks!

Hi, @ericspod. 
Regarding the comment "We already reference a EXAONEPath model hosted on hf in hf_models, if this is a later version of the same model perhaps it should be hosted in hf instead of here and we add a reference in hf_models in a similar pattern", this actually aligns well with our original intention.

We have already uploaded the EXAONE Path 2.0 model to Hugging Face at https://huggingface.co/LGAI-EXAONE/EXAONE-Path-2.0 (different from the past model), and we would like this version to be referenced in hf_models following the same pattern as the existing EXAONE Path model.

However, since we were unsure of the correct process to request this kind of update, we submitted the current pull request. Please let us know exact way to refer to EXAONE Path 2.0.

Thank you again!

ericspod commented:
> > Hi @cjs6211 thank you for the contribution. We do require that models in our zoo use the [bundle format](https://docs.monai.io/en/stable/bundle_intro.html). You have a lot of your own code here so I wouldn't suggest reformatting everything to use MONAI components, but there does need to be at least an inference config file which can call into code you have defined. This [bundle here](https://github.com/ericspod/model-zoo/tree/dev/models/brain_image_synthesis_latent_diffusion_model) is an example of an inference script with few components, and just calls into a defined method to implement an inference process. You can follow that pattern here but please do look at other bundles and see what you can do to make your code into a bundle.
> > We do also require that this model be in the `models` directory and not `hf_models` which contains information on those models hosted on Huggingface. We already reference a [EXAONEPath model hosted on hf](https://huggingface.co/LGAI-EXAONE/EXAONEPath) in `hf_models`, if this is a later version of the same model perhaps it should be hosted in hf instead of here and we add a reference in `hf_models` in a similar pattern.
> > @aylward could you please double check that this model's license is acceptable to us, I think it's fine from our previous discussions on licenses.
> > Thanks!
> 
> Hi, @ericspod. Regarding the comment "We already reference a EXAONEPath model hosted on hf in hf_models, if this is a later version of the same model perhaps it should be hosted in hf instead of here and we add a reference in hf_models in a similar pattern", this actually aligns well with our original intention.
> 
> We have already uploaded the EXAONE Path 2.0 model to Hugging Face at https://huggingface.co/LGAI-EXAONE/EXAONE-Path-2.0 (different from the past model), and we would like this version to be referenced in hf_models following the same pattern as the existing EXAONE Path model.
> 
> However, since we were unsure of the correct process to request this kind of update, we submitted the current pull request. Please let us know exact way to refer to EXAONE Path 2.0.
> 
> Thank you again!

Hi @cjs6211 I would suggest removing the files that you've proposed here and instead create a directory in `hf_models` with the contents similar to [the existing model](https://github.com/Project-MONAI/model-zoo/tree/dev/hf_models/exaonepath). You will want to replace the contents of the README.md and metadata.json files to match what your new model has but otherwise follow this as a template. The "model" link should point to the Huggingface link instead of the Github page however. I think that will work for what we want to host here and we can review again once that's done. Thanks!

cjs6211 committed:
Delete hf_models/EXAONE-Path-2.0 directory

Signed-off-by: Jongseong Jang <37769148+cjs6211@users.noreply.github.com>

cjs6211 committed:
Merge branch 'Project-MONAI:dev' into 757-add_exaone_path_2.0

cjs6211 committed:
Add files via upload

Signed-off-by: Jongseong Jang <37769148+cjs6211@users.noreply.github.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

cjs6211 committed:
Update README.md

Signed-off-by: Jongseong Jang <37769148+cjs6211@users.noreply.github.com>

cjs6211 committed:
Rename LICENSE.txt to LICENSE

Signed-off-by: Jongseong Jang <37769148+cjs6211@users.noreply.github.com>

cjs6211 commented:
> > > Hi @cjs6211 thank you for the contribution. We do require that models in our zoo use the [bundle format](https://docs.monai.io/en/stable/bundle_intro.html). You have a lot of your own code here so I wouldn't suggest reformatting everything to use MONAI components, but there does need to be at least an inference config file which can call into code you have defined. This [bundle here](https://github.com/ericspod/model-zoo/tree/dev/models/brain_image_synthesis_latent_diffusion_model) is an example of an inference script with few components, and just calls into a defined method to implement an inference process. You can follow that pattern here but please do look at other bundles and see what you can do to make your code into a bundle.
> > > We do also require that this model be in the `models` directory and not `hf_models` which contains information on those models hosted on Huggingface. We already reference a [EXAONEPath model hosted on hf](https://huggingface.co/LGAI-EXAONE/EXAONEPath) in `hf_models`, if this is a later version of the same model perhaps it should be hosted in hf instead of here and we add a reference in `hf_models` in a similar pattern.
> > > @aylward could you please double check that this model's license is acceptable to us, I think it's fine from our previous discussions on licenses.
> > > Thanks!
> > 
> > 
> > Hi, @ericspod. Regarding the comment "We already reference a EXAONEPath model hosted on hf in hf_models, if this is a later version of the same model perhaps it should be hosted in hf instead of here and we add a reference in hf_models in a similar pattern", this actually aligns well with our original intention.
> > We have already uploaded the EXAONE Path 2.0 model to Hugging Face at https://huggingface.co/LGAI-EXAONE/EXAONE-Path-2.0 (different from the past model), and we would like this version to be referenced in hf_models following the same pattern as the existing EXAONE Path model.
> > However, since we were unsure of the correct process to request this kind of update, we submitted the current pull request. Please let us know exact way to refer to EXAONE Path 2.0.
> > Thank you again!
> 
> Hi @cjs6211 I would suggest removing the files that you've proposed here and instead create a directory in `hf_models` with the contents similar to [the existing model](https://github.com/Project-MONAI/model-zoo/tree/dev/hf_models/exaonepath). You will want to replace the contents of the README.md and metadata.json files to match what your new model has but otherwise follow this as a template. The "model" link should point to the Huggingface link instead of the Github page however. I think that will work for what we want to host here and we can review again once that's done. Thanks!

Hi, @ericspod. Thanks for the guidance. I’ve removed the directory under hf_models and created a new exaonepath_2.0 directory instead. I followed the structure of the existing model, added the necessary files, and updated the contents of README.md and metadata.json to reflect the new model. Everything has been uploaded and should now be ready for review.

Best,

ericspod reviewed (approved):
I'm good with this addition now, thanks!

yiheng-wang-nv commented:
/build

yiheng-wang-nv commented:
/build
